### PR TITLE
Add some tests for the basic CSP configuration

### LINKED ISSUE #1379: Add some tests for CSP headers being set correctly
Relates #995

We should add some tests to make sure important headers are being set.

For example for script-src:

* Make sure unsafe-eval isn't present.
* Make sure unsafe-inline isn't present.

Check the default settings have the important hosts etc.

Check default-src is specified.

### PR DESCRIPTION
**Pull request by muffinresearch**
Fixes #1379

### CODE REVIEW

File: apps/amo/tests/test_csp_headers.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
+from django.conf import settings
+
+import lib.settings_base as base_settings
+
+import amo.tests
+import mock
+
+
+@mock.patch.object(settings, 'CSP_REPORT_ONLY', False)
```
Comment (magopian) [inline apps/amo/tests/test_csp_headers.py]: You can use [django.test.utils.override_settings](https://docs.djangoproject.com/en/1.9/topics/testing/tools/#django.test.override_settings) instead. Do we need that for the csp to be added to the response?

File: apps/amo/tests/test_csp_headers.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+class TestCSPHeaders(amo.tests.TestCase):
+
+    def test_for_specific_csp_settings(self):
+        """Test that require settings are provided as headers."""
```
Comment (magopian) [inline apps/amo/tests/test_csp_headers.py]: Typo? `Test that requireD settings...`?

File: apps/amo/tests/test_csp_headers.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        assert "report-uri" in response['content-security-policy']
+        # Basic assertions on the things we've defined.
+        assert "script-src" in response['content-security-policy']
+        assert "style-src" in response['content-security-policy']
+        assert "font-src" in response['content-security-policy']
+        assert "frame-src" in response['content-security-policy']
+
+    def test_unsafe_inline_not_in_script_src(self):
+        """Make sure a script-src does not have unsafe-inline."""
+        assert "'unsafe-inline'" not in base_settings.CSP_SCRIPT_SRC
```
Comment (magopian) [inline apps/amo/tests/test_csp_headers.py]: Would you rather check that it's not in `response['content-security-policy']` instead?

File: apps/amo/tests/test_csp_headers.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        for val in base_settings.CSP_STYLE_SRC:
+            assert not val.startswith('http:')
+
+    def test_http_protocol_not_in_img_src(self):
+        """Make sure a img-src does not have hosts using http:."""
+        for val in base_settings.CSP_IMG_SRC:
+            assert not val.startswith('http:')
+
+    def test_prod_cdn_in_common_settings(self):
+        """Make prod cdn is specified by default for statics."""
```
Comment (magopian) [inline apps/amo/tests/test_csp_headers.py]: Missing the word `sure`

**Comment by magopian** [comment]
this is very good, thanks!

r+wc

File: apps/amo/tests/test_csp_headers.py (hunk 3 above)
Comment (muffinresearch) [inline apps/amo/tests/test_csp_headers.py]: I would, but then I'd need to start parsing the header. As django-csp has tests and this is more direct it seemed like a better bet.

Comment (magopian) [inline apps/amo/tests/test_csp_headers.py]: fair enough ;)

**Pull request merged**